src/node_file_rimraf.cc

```diff
@@ -0,0 +1,111 @@
+#include "node_file.h"  // NOLINT(build/include_inline)
+#include "node_file-inl.h"
+
+#include <optional>
+
+#if defined(_WIN32)
+#include<windows.h>           // for windows
+#else
+#include<unistd.h>               // for linux
+#endif
+
+namespace node {
+
+namespace fs {
+
+using v8::FunctionCallbackInfo;
+using v8::Int32;
+using v8::Isolate;
+using v8::Value;
+
+#define TRACE_NAME(name) "fs.sync." #name
+#define GET_TRACE_ENABLED                                                      \
+  (*TRACE_EVENT_API_GET_CATEGORY_GROUP_ENABLED(                                \
+       TRACING_CATEGORY_NODE2(fs, sync)) != 0)
+#define FS_SYNC_TRACE_BEGIN(syscall, ...)                                      \
+  if (GET_TRACE_ENABLED)                                                       \
+    TRACE_EVENT_BEGIN(                                                         \
+        TRACING_CATEGORY_NODE2(fs, sync), TRACE_NAME(syscall), ##__VA_ARGS__);
+#define FS_SYNC_TRACE_END(syscall, ...)                                        \
+  if (GET_TRACE_ENABLED)                                                       \
+    TRACE_EVENT_END(                                                           \
+        TRACING_CATEGORY_NODE2(fs, sync), TRACE_NAME(syscall), ##__VA_ARGS__);
+
+static void UnlinkSync(char* path, uint32_t max_retries, uint32_t retry_delay) {
+  Isolate* isolate = Isolate::GetCurrent();
+  Environment* env = Environment::GetCurrent(isolate);
+
+  THROW_IF_INSUFFICIENT_PERMISSIONS(
+      env, permission::PermissionScope::kFileSystemWrite, path);
+
+  const auto tries = max_retries + 1;
+  constexpr std::array<int, 5> retryable_errors = {
+      EBUSY, EMFILE, ENFILE, ENOTEMPTY, EPERM};
+
+  uv_fs_t req;
+
+  for (uint64_t i = 1; i <= tries; i++) {
+    FS_SYNC_TRACE_BEGIN(unlink);
```

Comment:
Second parameter should be *path

Reply:
FS_SYNC_TRACE_BEGIN(unlink, *path) seems to fail with too many args error. Did I make a mistake here?
```
../../src/node_file_rimraf.cc:96:3: error: no matching function for call to 'AddTraceEvent'
  FS_SYNC_TRACE_BEGIN(stat, path);
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/node_file_rimraf.cc:28:5: note: expanded from macro 'FS_SYNC_TRACE_BEGIN'
    TRACE_EVENT_BEGIN(                                                         \
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/tracing/trace_event_common.h:280:3: note: expanded from macro 'TRACE_EVENT_BEGIN'
  INTERNAL_TRACE_EVENT_ADD(TRACE_EVENT_PHASE_BEGIN, category_group, name, \
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/tracing/trace_event.h:179:7: note: expanded from macro 'INTERNAL_TRACE_EVENT_ADD'
      node::tracing::AddTraceEvent(                                          \
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/tracing/trace_event.h:584:24: note: candidate function not viable: requires 7 arguments, but 8 were provided
static inline uint64_t AddTraceEvent(char phase,
                       ^
../../src/tracing/trace_event.h:595:24: note: candidate function template not viable: requires 9 arguments, but 8 were provided
static inline uint64_t AddTraceEvent(
                       ^
../../src/tracing/trace_event.h:609:24: note: candidate function template not viable: requires 11 arguments, but 8 were provided
static inline uint64_t AddTraceEvent(
```